### PR TITLE
AE-1791: upgraded megalinter to v6

### PR DESCRIPTION
**Pull request by Nomad-AE**
Copied over from Automation repo

### CODE REVIEW

File: .github/exemptions.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Reasons for ignoring certain checks
+
+## Megalinter Disabled Linters
+
+- jscpy - CopyPaste scanning is disabled
+- JAVASCRIPT_STANDARD - We use eslint instead
+
+## Specific linter exemptions
+
+- cspell - "ignoreRegExpList": "/[0-9a-f]{8}.js/gi" - scripts/downlaod.js will save files with this name
```
Comment (echoAE) [inline .github/exemptions.md:10]:
```suggestion
- cspell - "ignoreRegExpList": "/[0-9a-f]{8}.js/gi" - scripts/download.js will save files with this name
```

File: .github/exemptions.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- cspell - "ignoreRegExpList": "/[0-9a-f]{8}\\_ABI.json/gi" - scripts/downlaod.js will save files with this name
```
Comment (echoAE) [inline .github/exemptions.md:11]:
```suggestion
- cspell - "ignoreRegExpList": "/[0-9a-f]{8}\\_ABI.json/gi" - scripts/download.js will save files with this name
```

File: .github/exemptions.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- cspell - "ignoreRegExpList": "/0x[0-9a-f]{40}/gi" - ignore ETH addresses
+- cspell - "ignoreRegExpList": "/Qm[0-9a-zA-Z]{44}/g" - ignore IPFS CIDv0 addresses
+- cspell - "ignoreRegExpList": "/bafy[0-9a-zA-Z]{40}[0-9a-zA-Z]+/g" - ignore IPFS CIDv1 addresses
```
Comment (echoAE) [inline .github/exemptions.md:14]: See comments about IPFS CIDv1 format from comments here:
https://github.com/arbitraryexecution/automation-testing/pull/92

File: .github/gitleaks.toml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (echoAE) [inline .github/gitleaks.toml:1]: See comments from this pull request:
https://github.com/arbitraryexecution/automation-testing/pull/92

File: .github/workflows/mega-linter.yml [hunk 5]
```diff
@@ -35,23 +37,42 @@ jobs:
...
-          SPELL_CSPELL_DISABLE_ERRORS: true
-          # DISABLE_ERRORS: true # Uncomment if you want MegaLinter to detect errors but not block CI to pass
+
+          # eslint
+          JAVASCRIPT_ES_CONFIG_FILE: .github/eslintrc.js
+          # cspell settings
+          SPELL_CSPELL_CONFIG_FILE: .github/cspell.json
+          SPELL_CSPELL_DISABLE_ERRORS: true # Treat misspellings as a warning
+          # proselint settings
+          SPELL_PROSELINT_CONFIG_FILE: .github/proselint.json
```
Comment (echoAE) [inline .github/workflows/mega-linter.yml:62]:
```suggestion
          SPELL_PROSELINT_CONFIG_FILE: .github/proselintrc.json
```

File: .github/workflows/mega-linter.yml [hunk 6]
```diff
@@ -60,5 +81,35 @@ jobs:
         with:
           name: MegaLinter reports
           path: |
-            report
-            mega-linter.log
+            megalinter-reports
+
+      # Create pull request if applicable
+      # Note: This step will fail if megalinter tries to modify any file in the .github/workflow
+      # directory. Github has a protection that workflows cant modify workflows. Please fix those
```
Comment (echoAE) [inline .github/workflows/mega-linter.yml:88]:
```suggestion
      # directory. Github has a protection that workflows can't modify workflows. Please fix those
```

File: defender/downloaded/cToken_Monitor_ABI.json [hunk 7]
```diff
@@ -1 +1,1052 @@
...
\ No newline at end of file
+[
```
Comment (echoAE) [inline defender/downloaded/cToken_Monitor_ABI.json:1]: This is fine, although these ABI files are created programmatically if we pull them down from the Defender account.

I'll make a ticket for updating the ABI download to write them to file with the appropriate formatting.

**Review by echoAE — APPROVED** [review]

File: .github/exemptions.md (hunk 1 above)
Comment (Nomad-AE) [reply .github/exemptions.md]: addressed in e1250bd6ccf32489638864a367c380f8b73589cc

File: .github/exemptions.md (hunk 3 above)
Comment (Nomad-AE) [reply .github/exemptions.md]: addressed in e1250bd6ccf32489638864a367c380f8b73589cc

File: .github/gitleaks.toml (hunk 4 above)
Comment (Nomad-AE) [reply .github/gitleaks.toml:1]: addressed in e1250bd6ccf32489638864a367c380f8b73589cc

File: .github/workflows/mega-linter.yml (hunk 6 above)
Comment (Nomad-AE) [reply .github/workflows/mega-linter.yml]: addressed in e1250bd6ccf32489638864a367c380f8b73589cc

File: .github/exemptions.md (hunk 2 above)
Comment (Nomad-AE) [reply .github/exemptions.md]: addressed in e1250bd6ccf32489638864a367c380f8b73589cc

File: .github/cspell.json [hunk 8]
```diff
@@ -1,12 +1,11 @@
 {
   "ignorePaths": [
     "**/node_modules/**",
-    "**/vscode-extension/**",
     "**/.git/**",
-    ".vscode",
```
Comment (teasipper-ae) [inline .github/cspell.json:6]: Add .vscode back to list of ignored files?

**Review by teasipper-ae — APPROVED** [review]

**Pull request merged**